woocommerce/woocommerce-ios
# Product description AI: Update AI sheet UI

Part of #9937 


## Description

This PR updates the UI for the product description AI sheet following NMncL2ViLjomfLhuyEhVM1-fi. Changes include:
- Enabled the product name text field only if the product name is empty initially.
- Added focus state for the feature text field: the border is purple when the field is focused.
- Added a label for example features to add.
- Updated all wording "Write it for me" to "Write with AI".
- Moved the copy CTA to under the generated text.
- Removed the pause button when regenerating is in progress.
- Added loading state for the Regenerate button.
- Added skeleton for when regenerating is in progress.

Also: updated the "sparkles" icon in the tool bar of the description editor screen.

The legal text below the "Write with AI" button on the product detail screen will be added in a subsequent PR.

## Testing instructions

- Log in to a WPCom site.
- Switch to Products tab and select "+" to add a new product.
- Tap "Write with AI", and notice that the AI sheet is displayed with the correct design. The product name field should be editable.
- Navigate back and select an existing product and choose "Write with AI". The product name field should not be editable on the presented AI sheet.
- Tap "Write with AI" on the sheet. The UI should still be correct when generating and when a text is generated.
- Apply the description to get back to the product detail screen. Select the description row to open the product description editor.
- Notice that the entry point to AI on the keyboard toolbar is displayed with the correct "sparkles" icon.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/709f1eba-e6ac-43b1-9019-b8381b0598cc

Description editor:

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/3d5087c1-4fb2-4780-8ce7-6e43ecd2d391" width=320 />


---
